text/0000-module-resolution.md

```diff
@@ -0,0 +1,458 @@
+
+```json
+{
+    "modules": [
+        "just-string"
+    ]
+}
+```
+
+On the example below, it will be impossible to know if `just-string` is the name of an npm package, or a name of a folder that contains components, hence creationg resolution ambiguity.
+
+We considered other alternatives such as forcing specific prefix like `npm:package-name` or require a final slash (ex. `folder-name/`) for folders, but we believed that those were more cumbersome to learn and future-proof, so standarizing on an consistent object shape as ModuleRecord was a preferable and superior approach.
+
+### Types of ModuleRecord
+
+#### AliasModuleRecord
+
+This type of resolution maps a given module specifier to a particular path within the root of that module.
+
+```json
+{
+    "modules": [
+       {
+            "name": "ui/button",
+            "path": "src/modules/ui/button/button.js"
+        }        
+    ]
+}
+```
+
+`AliasModuleRecord` must contain `name` and a `path` keys.
+
+::: note 
+  It is encouraged that the name specifier for the modules is of the form namespace/name. Moreover the namespace doesn't necesarily have to match the path, altough it is recommended to follow that convention.
+:::
+
+#### DirectoryModuleRecord
+
+This type of resolution allows to specify a folder that contain LWC modules. The structure of the folder its very specific (LWC has a very opinionated module structure). It must contain a namespace folder, then a folder per module bundle (moduleName) and then a moduleEntry file that matches the name of the bundle (the entry must be of type .css, .html, .js or .ts)
+
+::: note
+  The folder structure is something that the core LWC team wants to enforce as a convention. There is no technical limitation on the compiler or other tools to resolve different folder structures. However we do want to provide and enforce as much as possible a canonical way so we can in the future keep iterating, improving and expanding the ergonomics of it.
+:::
+
+```json
+{
+    "modules": [
+        {
+           "dir" : "src/modules"
+        }
+    ]
+}
+```
+
+Here is an example of a valid module resolution given the previous DirectoryModuleRecord
+```
+src
+└── modules/ 
+    └── ui/ (namespace)
+        └── button/ (moduleName)
+            └── button.js (moduleEntry)
+```
+
+`DirectoryModuleRecord` must contain `dir` key.
+
+#### NpmModuleRecord
+
+This type of resolution tells the resolver to find an npm package with that name and resolve the modules it might have inside. This process can be recursive (see example at the top).
+
+```json
+{
+    "modules": [
+       { 
+           "npm" : "@ui/components"
+       }     
+    ]
+}
+```
+
+`NpmModuleRecord` must contain `npm` key.
+
+## Module resolution and exposure via NPM packages
+
+So far we have covered how are we going to resolve the modules, however we would want to be able to expose a subset of those modules 
+for other developers in the form of an npm package.
```

Comment:
a npm

Reply:
i think `an npm` is a more common phrase. see npm docs: https://docs.npmjs.com/configuring-your-registry-settings-as-an-npm-enterprise-user